R/drive_trash.R

```diff
@@ -0,0 +1,100 @@
+#' Trash file on Drive.
+#' @template file
+#' @template verbose
+#'
+#' @return Logical, indicating whether the trashing/untrashing succeeded.
+#' @export
+#' @examples
+#' \dontrun{
+#' ## Create a file to trash.
+#' file <- drive_upload(system.file("DESCRIPTION"), "DESC")
+#' ## Put a single file in the trash.
+#' drive_trash("DESC")
+#' drive_untrash("DESC")
+#'
+#' ## Put multiple files in trash.
+#' file_a <- drive_upload(system.file("DESCRIPTION"), "file_a")
+#' file_b <- drive_upload(system.file("DESCRIPTION"), "file_b")
+#' drive_trash(c("file_a", "file_b"))
+#' drive_untrash(c("file_a", "file_b"))
+#' }
+drive_trash <- function(file = NULL, verbose = TRUE) {
+  trash_file <- as_dribble(file)
+  if (!some_files(trash_file) && verbose) {
+    message(glue("No such files found to trash"))
+  }
+
+  out <- purrr::map_lgl(trash_file$id, trash_one)
+
+  if (verbose) {
+    if (any(out)) {
+      successes <- glue_data(trash_file[out, ], "  * {name}: {id}")
+      message(collapse(c("Files trashed:", successes), sep = "\n"))
+    }
+  }
+  invisible(out)
+}
+
+trash_one <- function(id, trash = TRUE) {
+  request <- generate_request(
+    endpoint = "drive.files.update",
+    params = list(fileId = id,
+                  trashed = trash)
+  )
+  response <- make_request(request, encode = "json")
+  proc_res <- process_response(response)
+  identical(proc_res$id, id)
+}
+
+#' @rdname drive_trash
+#' @export
+drive_untrash <- function(file = NULL, verbose = TRUE) {
+  if (is_path(file)) {
+    paths <- collapse(file, sep = "|")
+    file <- drive_find(paths, q = "trashed = true")
+  }
```

Comment:
It's true that you can pass q to `drive_find()`, but I think the proper way to do this so it could handle a path would be to pass it to `drive_paths()` which can't have a `q` passed at the moment.